hawkbit-repository/hawkbit-repository-jpa/src/test/java/org/eclipse/hawkbit/repository/jpa/DeploymentManagementTest.java

```diff
@@ -1352,11 +1354,79 @@ public void verifyDistributionSetAssignmentResultNotContainsAlreadyAssignedTarge
         assertThat(assignmentResult.getAssignedEntity()).hasSize(1);
 
         assignmentResult = assignDistributionSet(dsToTargetAssigned.getId(), savedTarget.getControllerId());
-        assertThat(assignmentResult.getAssignedEntity()).hasSize(0);
+        assertThat(assignmentResult.getAssignedEntity()).isEmpty();
 
         assertThat(distributionSetRepository.findAll()).hasSize(1);
     }
 
+    @Test
+    @Description("Verify that the DistributionSet assignments work for multiple targets of the same target type within the same request.")
+    void verifyDSAssignmentForMultipleTargetsWithSameTargetType() {
+        final DistributionSet ds = testdataFactory.createDistributionSet("test-ds");
+        final TargetType targetType = testdataFactory.createTargetType("test-type",
+                Collections.singletonList(ds.getType()));
+
+        final List<DeploymentRequest> deploymentRequests = new ArrayList<>();
+        for (int i = 0; i < 100; i++) {
+            final Target target1 = testdataFactory.createTarget("test-target-" + i, "test-target-" + i,
```

Comment:
`1` in name seems unnecessary

Reply:
Good catch, probably auto generated leftover 😅